src/Microsoft.AspNetCore.Cors/Infrastructure/UriHelpers.cs

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Cors.Infrastructure
+{
+    internal static class UriHelpers
+    {
+        public static bool IsSubdomainOf(Uri subdomain, Uri domain)
+        {
+            return subdomain.IsAbsoluteUri 
+                && domain.IsAbsoluteUri
+                && subdomain.Scheme == domain.Scheme
+                && subdomain.Port == domain.Port
+                && subdomain.Host.EndsWith($".{domain.Host}");
```

Comment:
This is pretty much how X509 certificates and HTTPS matching works. If you get a wildcard cert for *.contoso.com it doesn't match contoso.com. Mirroring this behaviour seems fine to me.